pkg/controller/workload/workload_processor.go

```diff
@@ -501,7 +501,7 @@ func (p *Processor) handleService(service *workloadapi.Service) error {
 	}
 
 	// Preprocess service, remove the waypoint from waypoint service, otherwise it will fall into a loop in bpf
-	if service.Waypoint != nil {
+	if service.Waypoint != nil && service.GetWaypoint().GetAddress() != nil && len(service.Addresses) != 0 {
```

Comment:
```suggestion
	if service.Waypoint != nil && len(service.GetWaypoint().GetAddress()) > 0 {
```

Reply:
The return type of `service.GetWaypoint().GetAddress()` is a struct, ref: https://github.com/kmesh-net/kmesh/blob/main/api/v2/workloadapi/workload.pb.go#L1292

Reply:
my fault, misread service.Addresses as waypoint.Address

Reply:
how could `service.Addresses` cause panic now